framework/include/fviks/FVInterfaceKernel.h

```diff
@@ -0,0 +1,113 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE
+#include "MooseObject.h"
+#include "SetupInterface.h"
+#include "ParallelUniqueId.h"
+#include "FunctionInterface.h"
+#include "DistributionInterface.h"
+#include "UserObjectInterface.h"
+#include "TransientInterface.h"
+#include "PostprocessorInterface.h"
+#include "VectorPostprocessorInterface.h"
+#include "GeometricSearchInterface.h"
+#include "BoundaryRestrictableRequired.h"
+#include "Restartable.h"
+#include "MeshChangedInterface.h"
+#include "TaggingInterface.h"
+#include "NeighborCoupleableMooseVariableDependencyIntermediateInterface.h"
+#include "TwoMaterialPropertyInterface.h"
+
+#include <set>
+
+class MooseMesh;
+class SubProblem;
+class FEProblemBase;
+class SystemBase;
+class Assembly;
+template <typename>
+class MooseVariableFV;
+
+/**
+ * Base class for creating kernels that interface physics between subdomains
+ */
+class FVInterfaceKernel : public MooseObject,
+                          public BoundaryRestrictableRequired,
+                          public SetupInterface,
+                          public FunctionInterface,
+                          public DistributionInterface,
+                          public UserObjectInterface,
+                          public TransientInterface,
+                          public PostprocessorInterface,
+                          public VectorPostprocessorInterface,
```

Comment:
That's quite a few more interfaces than FVKernel and InterfaceKernelBase
I understand this is planning for the future for some of these.

Is there an existing use for FE interface kernels using VPPs ? Distributions?

Reply:
I just added all the interfaces that are currently present in the union of `FVBoundaryCondition` and `FVFluxBC`. I do not know of any uses for VPPs. Would you rather I remove these until there is a use case?